src/main/java/com/microsoft/azure/functions/worker/Util.java

```diff
@@ -11,4 +16,19 @@ public static boolean isTrue(String value) {
     public static String getJavaVersion() {
         return String.join(" - ", System.getProperty("java.home"), System.getProperty("java.version"));
     }
+
+    public static void setGsonInstance(Gson instance) {
+        if (gsonInstance == null) {
+            synchronized (utilLock) {

```

Comment:
No need to add sync block here, it's already thread safe with the sync block at https://github.com/Azure/azure-functions-java-worker/blob/67258975a86bca1d7a356cad8c361ab79551a179/src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java#L69